src/os/src/detail/osTime.c

```diff
@@ -557,23 +557,29 @@ int64_t taosTimeSub(int64_t timeStamp, int64_t duration, char unit, int32_t prec
   return (int64_t)(mktime(&tm) * TSDB_TICK_PER_SECOND(precision));
 }
 
-int32_t taosTimeCountInterval(int64_t skey, int64_t ekey, int64_t interval, char unit, int32_t precision) {
+int64_t taosTimeTzOffset(int64_t intervalUnit, int32_t precision) {
+  int64_t tz_offset = 0;
+  if (intervalUnit == 'd' || intervalUnit == 'w') {
+    #if defined(WINDOWS) && _MSC_VER >= 1900
+    // see https://docs.microsoft.com/en-us/cpp/c-runtime-library/daylight-dstbias-timezone-and-tzname?view=vs-2019
+    int64_t timezone = _timezone;
+    #endif
+    tz_offset = -1 * timezone * TSDB_TICK_PER_SECOND(precision);
+  }
+  return tz_offset;
+}
+
+int32_t taosTimeCountInterval(int64_t skey, int64_t ekey, int64_t sliding, int64_t slidingUnit, int64_t intervalUnit, int32_t precision) {
```

Comment:
Refactor taosTimeCountInterval to respect the timezone offset according to intervalUnit, keeping in sync with behaviours of taosTimeTruncate.